### PR TITLE
NAS-122222 / 23.10 / UI should allow updating renew_days on cert update

### PR DESCRIPTION
**Pull request by AlexKarpov98**
for testing you need latest nightly and ACME certificate to be edited.

### CODE REVIEW

File: src/app/pages/credentials/certificates-dash/certificate-edit/certificate-edit.component.html [hunk 1]
```diff
@@ -15,6 +15,14 @@
             [required]="true"
             [tooltip]="helptext.edit.name.tooltip | translate"
           ></ix-input>
+          <ix-input
+            *ngIf="form.controls?.renew_days"
```
Comment (RehanY147) [inline src/app/pages/credentials/certificates-dash/certificate-edit/certificate-edit.component.html:19]: I feel like the `controls` array would always exist. Maybe this check should be on the form or not at all there.

File: src/app/pages/credentials/certificates-dash/certificate-edit/certificate-edit.component.ts [hunk 2]
```diff
@@ -54,19 +52,25 @@ export class CertificateEditComponent implements OnInit {
...
+  }
 
   get isCsr(): boolean {
     return this.certificate?.cert_type_CSR;
   }
 
   ngOnInit(): void {
     this.setCertificate();
+
+    if (this.certificate?.acme) {
```
Comment (RehanY147) [inline src/app/pages/credentials/certificates-dash/certificate-edit/certificate-edit.component.ts:66]: Lets make it a well named method like `setCertificate`

File: src/app/pages/credentials/certificates-dash/certificate-edit/certificate-edit.component.ts [hunk 3]
```diff
@@ -54,19 +52,25 @@ export class CertificateEditComponent implements OnInit {
     private errorHandler: FormErrorHandlerService,
     private matDialog: MatDialog,
     @Inject(SLIDE_IN_DATA) private slideInData: SlideInDataCertificateEdit,
-  ) {}
+  ) {
+    this.form.addControl('name', new FormControl(''));
```
Comment (RehanY147) [inline src/app/pages/credentials/certificates-dash/certificate-edit/certificate-edit.component.ts:56]: Why is this change needed? Need help understanding.

**Review by RehanY147 — CHANGES_REQUESTED** [review]

File: src/app/pages/credentials/certificates-dash/certificate-edit/certificate-edit.component.html (hunk 1 above)
Comment (AlexKarpov98) [reply src/app/pages/credentials/certificates-dash/certificate-edit/certificate-edit.component.html:19]: I do not check for array, I check for **renew_days** control presence. It is working, please recheck.

File: src/app/pages/credentials/certificates-dash/certificate-edit/certificate-edit.component.ts (hunk 2 above)
Comment (AlexKarpov98) [reply src/app/pages/credentials/certificates-dash/certificate-edit/certificate-edit.component.ts:66]: okay, gonna refactor it.

File: src/app/pages/credentials/certificates-dash/certificate-edit/certificate-edit.component.html (hunk 1 above)
Comment (RehanY147) [reply src/app/pages/credentials/certificates-dash/certificate-edit/certificate-edit.component.html:19]: The question mark `?` is checking whether the controls array exists. That's what I was talking about. Not the ngIf as a whole.

**Review by RehanY147 — CHANGES_REQUESTED** [review]
You can enter alphanumeric strings in the days input. It should have the 'numbers' type so you can only enter numbers.

**Comment by AlexKarpov98** [comment]
@RehanY147 - now it's all updated.

**Review by RehanY147 — APPROVED** [review]
